packages/grafana-data/src/types/queryRunner.ts

```diff
@@ -0,0 +1,36 @@
+import { Observable } from 'rxjs';
+import { DataQuery, DataSourceApi } from './datasource';
+import { PanelData } from './panel';
+import { ScopedVars } from './ScopedVars';
+import { TimeRange, TimeZone } from './time';
+
+/**
+ * Describes the options used when triggering a query via the {@link QueryRunner}.
+ *
+ * @internal
+ */
+export interface QueryRunnerOptions {
+  datasource: string | DataSourceApi | null;
+  queries: DataQuery[];
+  panelId?: number;
+  dashboardId?: number;
+  timezone: TimeZone;
+  timeRange: TimeRange;
+  timeInfo?: string; // String description of time range for display
+  maxDataPoints: number;
+  minInterval: string | undefined | null;
+  scopedVars?: ScopedVars;
+  cacheTimeout?: string;
+  app?: string;
```

Comment:
What's the reason for adding app here? Would alerting be considered an app since dashboard and explore is?

Also, should it not be of type `CoreApp` ?

Reply:
yes, app is for tracking/metrics. It cannot be of CoreApp if we want plugins to set it to their app name